maximum-cut/maximum_cut.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2019 D-Wave Systems, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## ------- Set up our graph -------
+import networkx as nx
+
+# Create empty graph
+G = nx.Graph()
+
+# Add edges to the graph (also adds nodes)
+G.add_edges_from([(1,2),(1,3),(2,4),(3,4),(3,5),(4,5)])
+
+## ------- Set up our QUBO dictionary -------
+
+# Initialize our Q matrix with all zeros
+Q = {}
+for u in G.nodes:
+    for v in G.nodes:
+        Q[(u,v)]=0
```

Comment:
Instead of initializing Q with zeros directly, we can use `defaultdict` which allow us to define the default value of our dictionary.
```
from collections import defaultdict

# Initialize our Q matrix with all zeros
# Note: int() returns 0. Hence, the defaultdict below
# assigns 0 to all keys that it has not seen before
Q = defaultdict(int)
```

For example,
```
In [1]: from collections import defaultdict                                                          

In [2]: Q = defaultdict(int)                                                                         

In [3]: Q[('a','b')] += 5        # Adding to never-seen-before key                                                                    

In [4]: Q                                                                                            
Out[4]: defaultdict(int, {('a', 'b'): 5})

In [5]: Q[('c')]        # Grabbing value for never-seen-before key                                                                             
Out[5]: 0

In [6]: Q                                                                                            
Out[6]: defaultdict(int, {('a', 'b'): 5, 'c': 0})
```